src/geowombat/core/sops.py

```diff
@@ -1116,66 +1127,103 @@ def coregister(target, reference, **kwargs):
         import geowombat as gw_
 
         if not AROSICS_INSTALLED:
-
             logger.exception('\nAROSICS must be installed to co-register data.\nSee https://pypi.org/project/arosics for details')
             raise NameError
 
         if isinstance(reference, str):
-
             if not os.path.isfile(reference):
-
                 logger.exception('  The reference file does not exist.')
                 raise OSError
 
             with gw_.open(reference) as reference:
                 pass
 
         if isinstance(target, str):
-
             if not os.path.isfile(target):
-
                 logger.exception('  The target file does not exist.')
                 raise OSError
 
             with gw_.open(target) as target:
                 pass
 
-        cr = arosics.COREG(reference.filename,
-                           target.filename,
-                           **kwargs)
-
-        try:
-            cr.calculate_spatial_shifts()
-        except:
+        with tempfile.TemporaryDirectory() as tmp:
+            ref_path = Path(tmp) / '_reference.tif'
+            tar_path = Path(tmp) / '_target.tif'
+            with gw_.open(reference.filename, chunks={'band': -1, 'y': 256, 'x': 256}) as ref_src:
+                if wkt_version is not None:
+                    ref_src = ref_src.assign_attrs(
+                        {
+                            'crs': ref_src.gw.crs_to_pyproj.to_wkt(version=wkt_version)
+                        }
+                    )
+                if 'nodata' in kwargs:
+                    ref_src = ref_src.assign_attrs({'nodatavals': (kwargs['nodata'][0],)}).fillna(kwargs['nodata'][0])
+                ref_src.gw.save(ref_path, overwrite=True)
+            with gw_.open(target.filename, chunks={'band': -1, 'y': 256, 'x': 256}) as tar_src:
+                if wkt_version is not None:
+                    tar_src = tar_src.assign_attrs(
+                        {
+                            'crs': tar_src.gw.crs_to_pyproj.to_wkt(version=wkt_version)
+                        }
+                    )
+                if 'nodata' in kwargs:
+                    tar_src = tar_src.assign_attrs({'nodatavals': (kwargs['nodata'][1],)}).fillna(kwargs['nodata'][1])
+                tar_src.gw.save(tar_path, overwrite=True)
+
+            cr = arosics.COREG(
+                GeoArray(GeoArray(str(ref_path)), projection=ref_src.crs),

```

Comment:
The projection can only be set on a `geoarray.GeoArray` if another `GeoArray` is passed as the input. E.g.
```python
GeoArray(str(ref_path), projection=ref_src.crs)
```
would not pass the reference CRS but the following does.
```python
GeoArray(GeoArray(str(ref_path)), projection=ref_src.crs)
```